src/covid_model_seiir_pipeline/pipeline/regression/model/ode_fit.py

```diff
@@ -269,52 +272,30 @@ def create_start_end_date_df(self):
         return df_result
 
 
-@dataclass
-class ODEProcessInput:
-    df_dict: Dict
-    col_date: str
-    col_cases: str
-    col_pop: str
-    col_loc_id: str
-    col_lag_days: str
-    col_observed: str
-
-    alpha: Tuple
-    sigma: Tuple
-    gamma1: Tuple
-    gamma2: Tuple
-    solver_dt: float
-    day_shift: Tuple
-
-
 class ODEProcess:
     """ODE Process for multiple group.
     """
-    def __init__(self, input):
-        """Constructor of ODEProcess.
-
-        Args:
-            input: ODEProcessInput
-        """
-        self.df_dict = input.df_dict
-        self.col_date = input.col_date
-        self.col_cases = input.col_cases
-        self.col_pop = input.col_pop
-        self.col_loc_id = input.col_loc_id
-        self.col_lag_days = input.col_lag_days
-        self.col_observed = input.col_observed
-
-        self.solver_dt = input.solver_dt
+    def __init__(self, ode_input: ODEProcessInput):
```

Comment:
could have saved some churn by just adding the type annotation `input: ODEProcessInput`